src/coreclr/vm/method.hpp

```diff
@@ -1813,22 +1822,25 @@ class MethodDesc
         m_wFlags |= mdfHasNativeCodeSlot;
     }
 
-    inline bool IsAsyncThunkMethod()
+    inline bool IsAsyncHelperMethod() const
     {
         LIMITED_METHOD_DAC_CONTRACT;
         if (!HasAsyncMethodData())
             return false;
-        AsyncMethodType asyncType = GetAddrOfAsyncMethodData()->type;
-        return asyncType == AsyncMethodType::AsyncToTask || asyncType == AsyncMethodType::TaskToAsync;
+        auto asyncType = GetAddrOfAsyncMethodData()->type;
+        return asyncType == AsyncMethodKind::AsyncThunkHelper ||
+            asyncType == AsyncMethodKind::AsyncImplHelper;
     }
 
-    inline bool IsAsync2Method()
+    // We use "async2" for runtime async methods that return "Unwrapped" values (i.e. T instead of Task<T>)
+    // The type of promise is typically captured in a modreq.
+    // CONSIDER: We probably need a better name for the concept, but it is hard to beat shortness of "async2"
+    inline bool IsAsync2Method() const
```

Comment:
I don't see good reason to have both this and `IsAsyncHelperMethod`. It seems like one of them should just be removed.

Reply:
"Thunk" and "Helper" were the same thing before this change and then had to be sorted out on what we actually want in ever place where "IsThunk" was used.  

Now we have the same situation with "IsHelper" and "IsAsync2". They are the same, but fundamentally do not need to be. We may end up with a scenario where we have to specify soemthing that is an "IsAsync2" but not a helper. - perhaps for internal helpers like `UnsafeAwaitAwaiterFromRuntimeAsync`

There are quite a few calls to `IsAsync2Method` and even more to `IsAsyncHelperMethod`

I'd like to keep these two separate for now. It is easy to fold then into one when we completely sure.